core/src/data-generator/batch-replayer.ts

```diff
@@ -0,0 +1,185 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+import { Construct, Duration } from "@aws-cdk/core";
+import { RetentionDays } from "@aws-cdk/aws-logs";
+import { LambdaInvoke } from "@aws-cdk/aws-stepfunctions-tasks";
+import { JsonPath, Map, StateMachine, TaskInput } from "@aws-cdk/aws-stepfunctions";
+import * as lambda from "@aws-cdk/aws-lambda";
+import { SfnStateMachine } from "@aws-cdk/aws-events-targets";
+import { Rule, Schedule } from "@aws-cdk/aws-events";
+import { Policy, PolicyStatement } from "@aws-cdk/aws-iam";
+import path = require("path");
+import { Bucket } from "@aws-cdk/aws-s3";
+import { PartitionedDataset } from "../datasets/partitioned-dataset";
+
+export interface BatchReplayerProps {
+  readonly dataset: PartitionedDataset;
+  readonly frequency?: number;
+  readonly sinkBucket: Bucket;
+  readonly outputFileMaxSizeInBytes?: number;
+}
+
+/**
+ * Replay the data in the given PartitionedDataset.
+ * 
+ * It will dump files into the `sinkBucket` based on the given `frequency`.
+ * The computation is in a Step Function with two Lambda steps.
+ * 
+ * 1. resources/lambdas/find-file-paths
+ * Read the manifest file and output a list of S3 file paths within that batch time range
+ * 
+ * 2. resources/lambdas/write-in-batch
+ * Take a file path, filter only records within given time range, adjust the the time with offset to
+ * make it looks like just being generated. Then write the output to the `sinkBucket`
+ */
+export class BatchReplayer extends Construct {
+
+  /**
+   * Dataset used for replay
+   */
+  public readonly dataset: PartitionedDataset;
+
+  /**
+   * Frequency (in Seconds) of the replaying. The batch job will start
+   * for every given frequency and replay the data in that period
+   */
+  public readonly frequency: number;
+
+  /**
+   * Sink bucket where the batch replayer will put data in
+   */
+  public readonly sinkBucket: Bucket;
+
+  /**
+   * Maximum file size for each output file. If the output batch file is,
+   * larger than that, it will be splitted into multiple files that fit this size.
+   * 
+   * Default to 100MB (max value)
+   */
+  public readonly outputFileMaxSizeInBytes?: number;
+
+  constructor(scope: Construct, id: string, props: BatchReplayerProps) {
+    super(scope, id);
+
+    this.dataset = props.dataset;
+    this.frequency = props.frequency || 60;
+    this.sinkBucket = props.sinkBucket;
+    this.outputFileMaxSizeInBytes = props.outputFileMaxSizeInBytes || 100 * 1024 * 1024; //Default to 100 MB
+    
+
+    /**
+     * Find all paths within the time range from the manifest file
+     */
+    const findFilePathsFn = new lambda.DockerImageFunction(this, "findFilePathFn", {
```

Comment:
can we use PreBundledFunction?

Reply:
No, it exceeds Lambda package size. I was considering a Lambda layer but there is no official one.